packages/roosterjs-editor-plugins/lib/plugins/ContentEdit/features/markdownFeatures.ts

```diff
@@ -172,6 +181,73 @@ const MarkdownInlineCode: BuildInEditFeature<PluginKeyboardEvent> = generateBasi
     false /* useShiftKey */
 );
 
+const RemoveCodeWhenEnterOnEmptyLine: BuildInEditFeature<PluginKeyboardEvent> = {
+    keys: [Keys.ENTER],
+    shouldHandleEvent: (event, editor) => {
+        const childOfCode = cacheGetCodeChild(event, editor);
+        return childOfCode && isNodeEmpty(childOfCode);
+    },
+    handleEvent: (event, editor) => {
+        event.rawEvent.preventDefault();
+        editor.addUndoSnapshot(
+            () => {
+                splitCode(event, editor);
+            },
+            null /* changeSource */,
+            true /* canUndoByBackspace */
+        );
+    },
+};
+
+const RemoveCodeWhenBackspaceOnEmptyFirstLine: BuildInEditFeature<PluginKeyboardEvent> = {
+    keys: [Keys.BACKSPACE],
+    shouldHandleEvent: (event, editor) => {
+        const childOfCode = cacheGetCodeChild(event, editor);
+        return childOfCode && isNodeEmpty(childOfCode) && !childOfCode.previousSibling;
+    },
+    handleEvent: (event, editor) => {
+        event.rawEvent.preventDefault();
+        editor.addUndoSnapshot(() => splitCode(event, editor));
+    },
+};
+
+function cacheGetCodeChild(event: PluginKeyboardEvent, editor: IEditor): Node | null {
+    return cacheGetEventData(event, 'CODE_CHILD', () => {
+        const codeElement = editor.getElementAtCursor('code');
+        if (codeElement) {
+            const pos = editor.getFocusedPosition();
+            const block = pos && editor.getBlockElementAtNode(pos.normalize().node);
+            if (block) {
+                const node =
+                    block.getStartNode() == codeElement.parentNode
+                        ? block.getStartNode()
+                        : block.collapseToSingleElement();
+                return isNodeEmpty(node) ? node : null;
+            }
+        }
+
+        return null;
+    });
+}
+
+function splitCode(event: PluginKeyboardEvent, editor: IEditor) {
+    const currentContainer = cacheGetCodeChild(event, editor);
+    if (!safeInstanceOf(currentContainer, 'HTMLElement')) {
+        return;
+    }
+    const codeChild = currentContainer.querySelector('code');
```

Comment:
Not sure if I understand the question
Currently for 
```html
<pre>
  <code>
    <...other divs>
    <div>
      <span>
        [selection here]
        <br>
      </span>
    </div>
    <...other divs>
   </code>
</pre>
```
Its split into 
```html
<pre>
   <code>
     <...other divs>
   </code>
   <code>
     <div>
       <span w selection>
     </div>
   <code>
   <other code>
</pre>
```
The selected `<code>` is unwrapped
And then this is split into three `<pre>`, selected one being unwrapped as well, leaving the new block under the editor div

Reply:
For this example, the expected result should be:

```html
<pre>
   <code>
     <...other divs>
   </code>
</pre>
<div>
    <span w selection>
</div>
<pre>
   <code>
   <other code>
</pre>
```

So both PRE and CODE tags are splitted.

This is because PRE tag has a built-in font that makes the text looks like code. So we need to move the text out of such styles.